### PR TITLE
Support for additional tags like <kbd> or <samp>

### PR DESCRIPTION
**Pull request by zeitgeist87**
This patch adds the attribute 'data-tags' to the script tag, to
allow the user to use additional tags like <kbd> or <samp>.

### CODE REVIEW

File: components/prism-core.js [hunk 1]
```diff
@@ -417,6 +424,7 @@ script = script[script.length - 1];
 
 if (script) {
 	_.filename = script.src;
+	_.tags = (script.getAttribute('data-tags') || 'code').split(/\s*,\s*/g);
```
Comment (LeaVerou) [inline components/prism-core.js]: If they're just tag names why comma separate them? Why not just space-separated?

File: components/prism-core.js [hunk 2]
```diff
@@ -146,7 +146,14 @@ var _ = _self.Prism = {
 	plugins: {},
 	
 	highlightAll: function(async, callback) {
-		var elements = document.querySelectorAll('code[class*="language-"], [class*="language-"] code, code[class*="lang-"], [class*="lang-"] code');
+		var selector = [];
+		for (var i = 0, tag; tag = _.tags[i++];) {
+			selector.push(tag + '[class*="language-"]');
+			selector.push('[class*="language-"] ' + tag);
+			selector.push(tag + '[class*="lang-"]');
+			selector.push('[class*="lang-"] ' + tag);
```
Comment (LeaVerou) [inline components/prism-core.js]: You don't need multiple push calls. push accepts multiple arguments.

**Comment by LeaVerou** [comment]
Interesting idea, thanks! See comments. Also, merge conflicts.

Also, it might produce cleaner code to do something like:

```js
var selector = _.tags.map(function(tag) {
	return [
		tag + '[class*="language-"]',
		'[class*="language-"] ' + tag,
		tag + '[class*="lang-"]',
		'[class*="lang-"] ' + tag
	].join(", ");
}).join(", ");
```

Just an idea, you don't have to follow it, but the current version of the code is a bit verbose and repetitive.

File: components/prism-core.js (hunk 1 above)
Comment (zeitgeist87) [inline components/prism-core.js]: Good point.

**Comment by zeitgeist87** [comment]
I have just updated the PR according to your suggestions. I changed the map function a little bit.

File: components/prism-core.js [hunk 3]
```diff
@@ -415,6 +420,8 @@ var script = document.currentScript || [].slice.call(document.getElementsByTagNa
 
 if (script) {
 	_.filename = script.src;
+	_.tags = (script.getAttribute('data-tags') || 'code').split(/\s+/g)
+				.filter(function(tag) {return !!tag;});
```
Comment (LeaVerou) [inline components/prism-core.js]: Why the filtering?

**Comment by LeaVerou** [comment]
Interesting idea.

However, I'm a bit worried that we're jumping to add features way too early, which is a slippery slope and can result in bloat over time. I actually like this feature and how elegantly we implemented it. However, the guy who actually opened #867 did not and I'm a bit hesitant to merge a feature we have no real use cases for. What do you think? :/

File: components/prism-core.js (hunk 3 above)
Comment (zeitgeist87) [inline components/prism-core.js]: To filter empty strings. For example the following code
```js
 '   code kbd   '.split(/\s+/g);
```
returns this array:
```js
[ '', 'code', 'kbd', '' ]
```
after the filter I get:
```js
[ 'code', 'kbd' ]
```

**Comment by zeitgeist87** [comment]
I totally agree. If nobody uses it, we shouldn't add it. It is only really useful with a custom theme anyway.

File: components/prism-core.js (hunk 3 above)
Comment (LeaVerou) [inline components/prism-core.js]: Just call `.trim()` before splitting then :)

**Comment by LeaVerou** [comment]
I actually needed something similar yesterday, and neither this, nor selector cut it: I was using `<script type="text/code">` to be able to have unescaped markup. Since it wasn't highlighted by default, I wrote a small script:

```js
$$("script[type='text/code']").forEach(script=>{
	var pre = document.createElement("pre");
	var code = document.createElement("code");
	pre.appendChild(code);
	script.parentNode.insertBefore(pre, script);
	code.appendChild(script);
});
```

It was better for me than replicating the entire selector just to add `script[type="text/code"]` to it.

**Pull request closed without merging**